File: Java/Programs/Product.java
// BAse class
class Product {
    // atribut produk protected agar dapat digunakan oleh keturunannya & no acces langsung(kalau beda package)
    protected String name; // name prod
    protected String id; // id prod
    protected String brands; // prod name
    protected double price; // prod price
    // construct kosong
    protected Product(){
    }
    //construct dengan value
    protected Product(String id, String name, String brands, double price){
        this.id = id;
        this.name = name;
        this.brands =brands;
        this.price = price;
    }
    // set void
    void set_id(String id){
        this.id = id;
    }
    void set_brands(String brands){
        this.brands = brands;
    }
    void set_name(String name){
        this.name = name;
    }
    void set_id(double price){
        this.price =price ;
    }
    // get methods
    String get_id(){
        return this.id;
    }
    String get_name(){
        return this.name;
    }
    String get_brands(){
        return this.brands;
    }
    double get_price(){
        return this.price;
    }
    // Destructor not needed in Java (Garbage Collection takes care of memory management)
}

File: Python/Programs/Product.py

# Saya Mohammad Faridz Fathin [2202680] mengerjakan LP2 dalam mata kuliah DPBO untuk keberkahanNya maka saya 
# tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
# Base Class
class Product:
    # Constructor
    def __init__(self, id="", name="", brands="", price=0):
        # Initialize attributes with default or provided values
        self.id = id
        self.name = name
        self.brands = brands
        self.price = price

    # Setter methods

    # Set the product ID
    def set_id(self, id):
        self.id = id

    # Set the product brand
    def set_brands(self, brands):
        self.brands = brands

    # Set the product name
    def set_name(self, name):
        self.name = name

    # Set the product price
    def set_price(self, price):
        self.price = price

    # Getter methods

    # Get the product ID
    def get_id(self):
        return self.id

    # Get the product name
    def get_name(self):
        return self.name

    # Get the product brand
    def get_brands(self):
        return self.brands

    # Get the product price
    def get_price(self):
        return self.price

    # Destructor (not explicitly needed in Python, handled automatically)
